### PR TITLE
Cjs upgrade

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

Upgrade cjs with a patch to use newer mozjs

// cc @worldofpeace @doronbehar I took the spidermonkey 78 commit from the GNOME 3.38 pr

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/cinnamon/cjs/default.nix [hunk 1]
```diff
@@ -17,26 +15,51 @@
...
-in
-
 stdenv.mkDerivation rec {
   pname = "cjs";
-  version = "4.6.0";
+  version = "4.6.0-gjs-1.66.0";
 
   src = fetchFromGitHub {
-    owner = "linuxmint";
+    owner = "leigh123linux";
```
Comment (doronbehar) [inline pkgs/desktops/cinnamon/cjs/default.nix:34]: What's wrong with upstream?

Comment (doronbehar) [reply pkgs/desktops/cinnamon/cjs/default.nix:34]: You should grab https://github.com/linuxmint/cjs/pull/84 as a patch.

**Comment by doronbehar** [comment]
@worldofpeace are we interested in having a `./generic.nix` for all the spidermonkey derivations?

**Comment by worldofpeace** [comment]
> @worldofpeace are we interested in having a `./generic.nix` for all the spidermonkey derivations?

It would be best to not duplicate them. But I think at some point they diverged greatly, and there's already code committed here. So I would say a bug for "later". And, yeah, I've thought about this too.

File: pkgs/desktops/cinnamon/cjs/default.nix (hunk 1 above)
Comment (mkg20001) [reply pkgs/desktops/cinnamon/cjs/default.nix:34]: The patch is quite gigantic. There have been some problems with renames not being applied in previous patches I applied, so I decided to use that approach as it seems saner.

**Review by worldofpeace — APPROVED** [review]

**Comment by quyse** [comment]
`gjs-1.66.0` of `leigh123linux/cjs` repo is a branch, not a tag, and there were a few commits to that branch since then. At the moment fresh fetching of `cjs` source is broken because of wrong `sha256`: it should be now `00a7zq5vvz65n18apdzqaai4pjzv26zybkxdspy17xcdyl9rk497`. `rev` should probably be pointed to a specific commit, not a branch?

**Comment by mkg20001** [comment]
Oh yes, my bad

**Comment by worldofpeace** [comment]
Oh goodness didn't notice that. It would be cool if CI checks wouldn't allow such unstable references in fetchers.

**Pull request merged**